engine/rpcserver.go

```diff
@@ -1844,3 +1887,265 @@ func (s *RPCServer) GCTScriptAutoLoadToggle(ctx context.Context, r *gctrpc.GCTSc
 	}
 	return &gctrpc.GCTScriptGenericResponse{Status: "success", Data: "script " + r.Script + " added to autoload list"}, nil
 }
+
+// SetExchangeAsset enables or disables an exchanges asset type
+func (s *RPCServer) SetExchangeAsset(_ context.Context, r *gctrpc.SetExchangeAssetRequest) (*gctrpc.GenericSubsystemResponse, error) {
+	exch := GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errors.New("exchange is not loaded/doesn't exist")
+	}
+
+	exchCfg, err := Bot.Config.GetExchangeConfig(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	base := exch.GetBase()
+	if base == nil {
+		return nil, errors.New("cannot get exchange base")
+	}
+
+	if r.Asset == "" {
+		return nil, errors.New("asset type must be specified")
+	}
+
+	a := asset.Item(r.Asset)
+
+	if r.Enable {
+		err = base.CurrencyPairs.EnableAsset(a)
+		if err != nil {
+			return nil, err
+		}
+		err = exchCfg.CurrencyPairs.EnableAsset(a)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		err = base.CurrencyPairs.DisableAsset(a)
+		if err != nil {
+			return nil, err
+		}
+		err = exchCfg.CurrencyPairs.DisableAsset(a)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &gctrpc.GenericSubsystemResponse{}, nil
+}
+
+// SetAllExchangePairs enables or disables an exchanges pairs
+func (s *RPCServer) SetAllExchangePairs(_ context.Context, r *gctrpc.SetExchangeAllPairsRequest) (*gctrpc.GenericSubsystemResponse, error) {
+	exch := GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errors.New("exchange is not loaded/doesn't exist")
+	}
+
+	exchCfg, err := Bot.Config.GetExchangeConfig(r.Exchange)
+	if err != nil {
+		return nil, err
+	}
+
+	base := exch.GetBase()
+	if base == nil {
+		return nil, errors.New("cannot get exchange base")
+	}
+
+	assets := base.CurrencyPairs.GetAssetTypes()
+
+	if r.Enable {
+		for i := range assets {
+			var pairs currency.Pairs
+			pairs, err = base.CurrencyPairs.GetPairs(assets[i], false)
+			if err != nil {
+				return nil, err
+			}
+			exchCfg.CurrencyPairs.StorePairs(assets[i], pairs, true)
+			base.CurrencyPairs.StorePairs(assets[i], pairs, true)
+		}
+
+		err = exch.FlushWebsocketChannels()
```

Comment:
I got the following output from running a command:
`.\gctcli.exe pair enableall --exchange=gateio
2020/06/30 14:57:54 rpc error: code = Unknown desc = websocket service not enabled`

The command is successful, but still returns an error, however this code shouldn't be reached if the websocket is already disabled. So please add a check to see if the websocket is enabled before executing this, otherwise a user will receive an error that's not really an error.

Please also do this for `UpdateExchangeSupportedPairs` and `UpdateExchangeSupportedPairs`